pkg/solana/chainwriter/transform_registry.go

```diff
@@ -2,49 +2,105 @@ package chainwriter
 )
 
 type ReportPostTransform struct {
-	ReportContext  [2][32]byte
-	Report         []byte
-	Info           ccipocr3.ExecuteReportInfo
-	AbstractReport ccip_offramp.ExecutionReportSingleChain
-	TokenIndexes   []byte
+	ReportContext [2][32]byte
+	Report        []byte
+	Info          ccipocr3.ExecuteReportInfo
+	TokenIndexes  []byte
+}
+
+type ExtraDataDecoded struct {
+	// ExtraArgsDecoded contain message specific extra args.
+	ExtraArgsDecoded map[string]any
+	// DestExecDataDecoded contain token transfer specific extra args.
+	DestExecDataDecoded []map[string]any
 }
 
-func FindTransform(id string) (func(context.Context, any, solana.AccountMetaSlice, map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, error), error) {
+type SVMExecCallArgs struct {
+	ReportContext [2][32]byte                `mapstructure:"ReportContext"`
+	Report        []byte                     `mapstructure:"Report"`
+	Info          ccipocr3.ExecuteReportInfo `mapstructure:"Info"`
+	ExtraData     ExtraDataDecoded           `mapstructure:"ExtraData"`
+}
+
+type SVMCommitCallArgs struct {
+	ReportContext [2][32]byte               `mapstructure:"ReportContext"`
+	Report        []byte                    `mapstructure:"Report"`
+	Rs            [][32]byte                `mapstructure:"Rs"`
+	Ss            [][32]byte                `mapstructure:"Ss"`
+	RawVs         [32]byte                  `mapstructure:"RawVs"`
+	Info          ccipocr3.CommitReportInfo `mapstructure:"Info"`
+}
+
+// TODO: replace with actual value from CCIP on-chain
+const StaticCuOverhead uint32 = 120000
+
+func FindTransform(id string) (func(context.Context, any, solana.AccountMetaSlice, map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, []txmutils.SetTxConfig, error), error) {
 	switch id {
 	case "CCIPExecute":
-		return CCIPExecuteArgsTransform, nil
+		return CCIPExecuteTransform, nil
 	case "CCIPCommit":
-		return CCIPCommitAccountTransform, nil
+		return CCIPCommitTransform, nil
 	default:
 		return nil, fmt.Errorf("transform not found")
 	}
 }
 
 // This Transform function looks up the token pool addresses in the accounts slice and augments the args
 // with the indexes of the token pool addresses in the accounts slice.
-func CCIPExecuteArgsTransform(ctx context.Context, args any, accounts solana.AccountMetaSlice, tableMap map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, error) {
-	var argsTransformed ReportPostTransform
-	err := mapstructure.Decode(args, &argsTransformed)
+func CCIPExecuteTransform(ctx context.Context, args any, accounts solana.AccountMetaSlice, tableMap map[string]map[string][]*solana.AccountMeta) (any, solana.AccountMetaSlice, []txmutils.SetTxConfig, error) {
+	var argsDecoded SVMExecCallArgs
+	err := mapstructure.Decode(args, &argsDecoded)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, []txmutils.SetTxConfig{}, err
+	}
+
+	argsTransformed := ReportPostTransform{
```

Comment:
Looks like the field names are the same, couldn't we keep this like it was before and just decode directly into `ReportPostTransform` rather than `SVMExecCallArgs`?

Reply:
I think `SVMExecCallArgs` is a better name, because we will be importing this type directly into the ccip contract transmitter, so it doesn't really make sense to have the transform name in there, plus this is more than just a report. We could potentially combine the two types. `SVMExecCallArgs` is currently missing `TokenIndexes` but it could just be not set in the transmitter and sent over to be populated, then `ExtraData` will be discarded when the args are encoded for chain.

Reply:
Went ahead and made those changes

Reply:
Ya agree `SVMExecCallArgs` is probably the better name. Just didn't want to mapdecode into a struct then parse the info into another. The changes look good!